pkg/server/server.go

```diff
@@ -222,15 +246,161 @@ func (s *Server) Run(ctx context.Context) error {
+	serverOptions.SecureServing.LoopbackKeyPEM = loopbackClientCertKey
+	serverOptions.SecureServing.LoopbackToken = uuid.New().String()
+	loopbackClientConfig, err := serverOptions.SecureServing.NewLoopbackClientConfig(serverOptions.SecureServing.LoopbackToken, serverOptions.SecureServing.LoopbackCertPEM)
+	if err != nil {
+		return err
+	}
+
+	// FIXME: switch to a single set of shared informers
+	const crossCluster = "*"
+	kcpClusterClient, err := kcpclient.NewClusterForConfig(loopbackClientConfig)
+	if err != nil {
+		return err
+	}
+	kcpClient := kcpClusterClient.Cluster(crossCluster)
+
+	kcpSharedInformerFactory := kcpexternalversions.NewSharedInformerFactoryWithOptions(kcpClient, resyncPeriod)
+	s.AddPostStartHook("FIXME-start-informers-for-crd-getter", func(context genericapiserver.PostStartHookContext) error {
+		kcpSharedInformerFactory.Start(context.StopCh)
+		return nil
+	})
+	// FIXME: (end) switch to a single set of shared informers
+
+	workspaceLister := kcpSharedInformerFactory.Tenancy().V1alpha1().Workspaces().Lister()
+
+	serverOptions.APIExtensionsNewSharedInformerFactoryFunc = func(client apiextensionsclient.Interface, resyncPeriod time.Duration) crdexternalversions.SharedInformerFactory {
+		f := apiextensionsinformers.NewSharedInformerFactory(client, resyncPeriod)
+		return &kcpAPIExtensionsSharedInformerFactory{
+			SharedInformerFactory: f,
+			workspaceLister:       workspaceLister,
+		}
+	}
+
 	cpOptions, err := genericcontrolplane.Complete(serverOptions)
 	if err != nil {
 		return err
 	}
 
-	server, err := genericcontrolplane.CreateServerChain(cpOptions, ctx.Done())
+	serverChain, err := genericcontrolplane.CreateServerChain(cpOptions, ctx.Done())
 	if err != nil {
 		return err
 	}
+	server := serverChain.MiniAggregator.GenericAPIServer
+
+	serverChain.GenericControlPlane.GenericAPIServer.Handler.GoRestfulContainer.Filter(func(req *restful.Request, res *restful.Response, chain *restful.FilterChain) {
+		ctx := req.Request.Context()
+		requestInfo, ok := apirequest.RequestInfoFrom(ctx)
+		if !ok {
+			responsewriters.ErrorNegotiated(
+				apierrors.NewInternalError(fmt.Errorf("no RequestInfo found in the context")),
+				// TODO is this the right Codecs?
+				scheme.Codecs, schema.GroupVersion{Group: requestInfo.APIGroup, Version: requestInfo.APIVersion}, res.ResponseWriter, req.Request,
+			)
+			return
+		}
+
+		// If it's not the core group, pass through
+		if requestInfo.APIGroup != "" {
+			chain.ProcessFilter(req, res)
+			return
+		}
+
+		// Special handling of the core ("") group
+
+		if requestInfo.IsResourceRequest {
+			// This is a CRUD request for somethig like pods. Try to see if there is a CRD for the resource. If so, let the CRD
+			// server handle it.
+			crdName := requestInfo.Resource + ".core"
+			_, err := serverChain.CustomResourceDefinitions.Informers.Apiextensions().V1().CustomResourceDefinitions().Lister().GetWithContext(ctx, crdName)
+			if err == nil {
```

Comment:
No error handling here (yet)... does it just make sense to log and still return a response?